src/tracer.js

```diff
@@ -177,21 +178,21 @@ export default class Tracer {
         fields.operationName = operationName;
 
         let tags = fields.tags || {};
-        let references = fields.references || [];
         let startTime = fields.startTime;
         if (!startTime) {
             startTime = Utils.getTimestampMicros();
         }
 
-        // TODO(oibe) support use of references
         let parent: ?SpanContext = fields.childOf instanceof Span ? fields.childOf.context(): fields.childOf;
-        if (!parent) {
+        if (!parent && fields.references) {
             // If there is no childOf in fields, then search list of references
-            for (let i = 0; i < references.length; i++) {
-                let ref: Reference = references[i];
+            for (let i = 0; i < fields.references.length; i++) {
+                let ref: Reference = fields.references[i];
                 let ref_type = ref.type();
                 if ((ref_type === opentracing.REFERENCE_CHILD_OF) ||
                     (ref_type === opentracing.REFERENCE_FOLLOWS_FROM)) {
+                    // TODO(oibe) which refference to use as parent when we have a list of
+                    // references? Should a new parent be created as when we receive no 'childOf'?
```

Comment:
NB: we do not need to keep the child_of reference in the span since we already represent it with the `parent` field.

NB: we should also guard against child_of reference that has a different trace ID - that would indicate a mistake in the instrumentation. I think we can convert such reference to FOLLOWS_FROM instead